src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -327,6 +329,8 @@ private ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, out Binde
                 }
             }
 
+            NullableWalker.AnalyzeIfNeeded(binder, parameterEqualsValue, diagnostics);
```

Comment:
I'd be interested in scoping out the work to make this happen in parameter default values as well as attribute arguments, since they appear to be similar in this respect.